libcontainer/restored_process.go

```diff
@@ -89,19 +90,19 @@ type nonChildProcess struct {
 }
 
 func (p *nonChildProcess) start() error {
-	return newGenericError(fmt.Errorf("restored process cannot be started"), SystemError)
+	return newGenericError(errors.New("restored process cannot be started"), SystemError)
 }
 
 func (p *nonChildProcess) pid() int {
 	return p.processPid
 }
 
 func (p *nonChildProcess) terminate() error {
-	return newGenericError(fmt.Errorf("restored process cannot be terminated"), SystemError)
+	return newGenericError(errors.New("restored process cannot be terminated"), SystemError)
```

Comment:
It would be great if we got rid of these `newGenericError()` etc. Afaics, these were created to provide the stack as part of the error, but `github.com/pkg/errors` provides the same.

Reply:
Yes, I agree, see my TODO item in this PR description above.

I took a quick look as to how dismantle all this and it seems it's easier to do it in a separate PR (once this one is merged).

Reply:
Yeah `newGenericError` came from a time before `pkg/error` existed and to be honest I've always hated the `genericError` error messages. But yeah, let's do it in a separate PR.